core/ssh_keys_test.go

```diff
@@ -0,0 +1,372 @@
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			testJSONMarshaling(t, tt.obj)
+		})
+	}
+}
+
+func TestNewSSHKeysClient(t *testing.T) {
+	tc := testclient.New(nil, nil, nil)
+	c := NewSSHKeysClient(tc)
+	assert.Equal(t, tc, c.client)
+	assert.Equal(t, &url.URL{Path: "/core/v1/"}, c.basePath)
+}
+
+func TestSSHKeysClient_List(t *testing.T) {
+	type args struct {
+		ctx context.Context
+		org OrganizationRef
+	}
+	tests := []struct {
+		name    string
+		args    args
+		resp    *katapult.Response
+		respErr error
+		respV   *sshKeysResponseBody
+		want    []*AuthSSHKey
+		wantReq *katapult.Request
+		wantErr string
+	}{
+		{
+			name: "success",
+			args: args{
+				ctx: context.Background(),
+				org: OrganizationRef{ID: "org_O648YDMEYeLmqdmn"},
+			},
+			resp: &katapult.Response{
+				Pagination: &katapult.Pagination{Total: 333},
+			},
+			respV: &sshKeysResponseBody{
+				SSHKeys: []*AuthSSHKey{
+					{ID: "ssh_O574YEEEYeLmqdmn"},
+				},
+			},
+			want: []*AuthSSHKey{
+				{ID: "ssh_O574YEEEYeLmqdmn"},
+			},
+			wantReq: &katapult.Request{
+				Method: "GET",
+				URL: &url.URL{
+					Path: "/core/v1/organizations/_/ssh_keys",
+					RawQuery: url.Values{
+						"organization[id]": []string{
+							"org_O648YDMEYeLmqdmn",
+						},
+					}.Encode(),
+				},
+			},
+		},
+		{
+			name: "request error",
+			args: args{
+				ctx: context.Background(),
+				org: OrganizationRef{ID: "org_O648YDMEYeLmqdmn"},
+			},
+			respErr: fmt.Errorf("flux capacitor undercharged"),
+			wantErr: "flux capacitor undercharged",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tc := testclient.New(tt.resp, tt.respErr, tt.respV)
+			c := NewSSHKeysClient(tc)
+			ctx := test.Context(tt.args.ctx)
+
+			got, resp, err := c.List(ctx, tt.args.org)
+
+			assert.Equal(t, 1, len(tc.Calls), "only 1 request should be made")
+			test.AssertContext(t, ctx, tc.Ctx)
+
+			assert.Equal(t, tt.want, got)
+
+			if tt.resp != nil {
+				assert.Equal(t, tt.resp, resp)
+			}
+
+			if tt.wantReq != nil {
+				assert.Equal(t, tt.wantReq, tc.Request)
+			}
+
+			if tt.wantErr == "" {
+				assert.NoError(t, err)
+			} else {
+				assert.EqualError(t, err, tt.wantErr)
+			}
+		})
+	}
+}
+
+//nolint:lll
+const sshPublicKey = "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQC7MZzjBzFWsc6BCcYE2EpSo8DOjzhDPb/WndW6QE/G0xM7iqdlezcmQnL3Gw9jtAOI4OlNok19v4q8C6ham+1WbX2aGd2labOmKoBVWXIzKyFz9pg2Rs10ZGn+Ly+uJF558rSehSvGJPFmKUagYeBG9c/cwuVMzube0yVbtH2CWRs2dMvwhloH5zOh3NMQj/5uBGYMh9uRQKsGHoG8TET08VSok3W/CFilSH7jSmaQYziUqJjOLE2hb8ziCzfv/0GhbY5MoJJUZqUdOlGkYgDMR/IVOHxxF93QBvp1AkAzh8RBsvJPajgZHFa1lWYJRP7U4TREWuxkpaJrbK3I3AHM74GAfIq76wndoFYJhi5qbNgaJjLUJDPPzl8KOcp0Pb5FPqygHWz/K4n1h5SV/LdD0mB487TxeC1NV4XBQQruM5RgfTXSWBW+8W83U0y5h1RNl/Qo9Efo7Kyc25wCxVT2cWRHr3mxZ98p+JxmFmC1KTdUrM95+B7+Hw9fKYvhKz0= jake@Jakes-MacBook-Pro.local"
+
+//nolint:lll
+const sshFingerprint = "SHA256:Ybk7/sbyptVqD87piCCz/XHiEKrdvHND2EMDA1qGqRA jake@Jakes-MacBook-Pro.local"
```

Comment:
Nitpick: For static things like this, I prefer breaking the value over multiple lines just to make it play nicer with editors and linters.

```suggestion
const sshPublicKey = "ssh-rsa " +
	"AAAAB3NzaC1yc2EAAAADAQABAAABgQC7MZzjBzFWsc6BCcYE2EpSo8DOjzhDPb/WndW6QE/G" +
	"0xM7iqdlezcmQnL3Gw9jtAOI4OlNok19v4q8C6ham+1WbX2aGd2labOmKoBVWXIzKyFz9pg2" +
	"Rs10ZGn+Ly+uJF558rSehSvGJPFmKUagYeBG9c/cwuVMzube0yVbtH2CWRs2dMvwhloH5zOh" +
	"3NMQj/5uBGYMh9uRQKsGHoG8TET08VSok3W/CFilSH7jSmaQYziUqJjOLE2hb8ziCzfv/0Gh" +
	"bY5MoJJUZqUdOlGkYgDMR/IVOHxxF93QBvp1AkAzh8RBsvJPajgZHFa1lWYJRP7U4TREWuxk" +
	"paJrbK3I3AHM74GAfIq76wndoFYJhi5qbNgaJjLUJDPPzl8KOcp0Pb5FPqygHWz/K4n1h5SV" +
	"/LdD0mB487TxeC1NV4XBQQruM5RgfTXSWBW+8W83U0y5h1RNl/Qo9Efo7Kyc25wCxVT2cWRH" +
	"r3mxZ98p+JxmFmC1KTdUrM95+B7+Hw9fKYvhKz0= jake@Jakes-MacBook-Pro.local"

const sshFingerprint = "SHA256:Ybk7/sbyptVqD87piCCz/XHiEKrdvHND2EMDA1qGqRA " +
	"jake@Jakes-MacBook-Pro.local"

```